diff --git a/rosbot_description/urdf/common/gazebo.urdf.xacro b/rosbot_description/urdf/common/gazebo.urdf.xacro
index cb44646..be7a5a6 100644
--- a/rosbot_description/urdf/common/gazebo.urdf.xacro
+++ b/rosbot_description/urdf/common/gazebo.urdf.xacro
@@ -22,6 +22,14 @@
           <remapping>/tf_static:=tf_static</remapping>
         </ros>
       </plugin>
+
+      <plugin filename="gz-sim-odometry-publisher-system" name="gz::sim::systems::OdometryPublisher">
+        <odom_publish_frequency>50</odom_publish_frequency>
+        <dimensions>2</dimensions>
+        <odom_topic>/model/rosbot/odometry_gt</odom_topic>
+        <odom_frame>odom_gt</odom_frame>
+        <robot_base_frame>base_link</robot_base_frame>
+      </plugin>
     </gazebo>
   </xacro:macro>
 
